docs/arch/adr-001-use-lettuce-core-lib-instead-of-dropwizard-redis.md

```diff
@@ -0,0 +1,34 @@
+# ADR 001 - Use lettuce-core library instead of dropwizard redis
+
+## Context
+
+There was an [intention](https://payments-platform.atlassian.net/browse/PP-6343) to use the 
+[dropwizard-redis](https://github.com/dropwizard/dropwizard-redis) library. The advantages of using this library are:
+
+* Configuration
+* Client lifecycle management
+* Client health checks
+* Dropwizard Metrics integration
+
+The dropwizard-redis library's io.dropwizard.redis.RedisClientFactory class 
+[expects](https://github.com/dropwizard/dropwizard-redis/blob/master/src/main/java/io/dropwizard/redis/RedisClientFactory.java#L54) 
+to be able to make a connection with Redis on application startup. This has some unintended consequences:
+
+* Mandating a connection on startup causes many integration tests to fail. Making a redis connection available for every relevant test is a fairly big change.
+* Mandating a connection might affect running Publicapi in dev/local environments.
+
+## Decision
+
+We will use the [lettuce-core](https://github.com/lettuce-io/lettuce-core) instead. Reasons for this are:
+
+* The dropwizard-library uses the lettuce-core libarary internally anyway
+* Publicapi is currently designed to be able to work without Redis
+* Redis in our environments is currently a single instance
```

Comment:
Why is this a reason to use `lettuce-core`?

Reply:
Yeah this isn't a valid reason. I originally thought this was a reason because even with a failover HA will failover to another instance but it would mean the data is wiped clean. But I think that's not correct.